### PR TITLE
[skip ci] Document CFLAGS and friends in main docs

### PR DESCRIPTION
**Pull request by flub**
Currently this is only partially documented in the quick-start guide,
include this in the main docs and document all the environment
argumens used.

### CODE REVIEW

**Comment by jpakkane** [comment]
Looks nice, but the actual table should go in the reference tables page instead and the manual page should just link to that (having the same information in two different places means that they will get out of sync). Thanks.

**Comment by flub** [comment]
Yeah, fair point.  I was considering this originally as well.  Updated.

**Comment by flub** [comment]
I added something along those lines, not 100% sure it's correct though so any further tips on clarifications welcome.

File: docs/markdown/Running-Meson.md [hunk 1]
```diff
@@ -92,6 +92,21 @@ You can then open the generated solution with Visual Studio and
...
+Environment Variables
+--
+
+Sometimes you want to add extra compiler flags, this can be done by
+passing them in environment variables when calling meson. See [the
+reference
+tables](Reference-tables.md#compiler-and-linker-flag-envrionment-variables)
+for a list of all the environment variables. Be aware however these
+environment variables are only used for native builds and will not
+affect the build when cross-compiling.
```
Comment (jpakkane) [inline docs/markdown/Running-Meson.md]: They do affect the build, but only for selecting the native compiler. The cross compiler always comes from the cross file.

Comment (xclaesse) [reply docs/markdown/Running-Meson.md]: Not only for selecting the native compiler, but also CFLAGS and friends are passed to native compiler too, AFAIK.

Comment (xclaesse) [reply docs/markdown/Running-Meson.md]: Unrelated: just realized 'native' keyword argument is not documented in reference manual for executable().

Comment (flub) [reply docs/markdown/Running-Meson.md]: Is the current wording acceptable?

**Comment by jpakkane** [comment]
Thanks.

**Pull request merged**